run.sh

```diff
@@ -0,0 +1,41 @@
+#!/bin/sh
+
+set -eu
+
+#set the max memory
+BYTES_PER_MEG=$((1024*1024))
+BYTES_PER_GIG=$((1024*${BYTES_PER_MEG}))
+
+DEFAULT_MIN=$((64 * $BYTES_PER_MEG)) #This is a guess
+regex='^([[:digit:]]+)([GgMm])?i?$'
+num_regex='^[[:digit:]]+'
+unit_regex='[GgMm]'
+
+export NODE_OPTIONS=""
+
+if [ 1 -eq $(echo "${OCP_AUTH_PROXY_MEMORY_LIMIT:-}" | grep -Ec $regex) ] ; then
+    num=$(echo ${OCP_AUTH_PROXY_MEMORY_LIMIT} | grep -oE $num_regex)
+    unit=$(echo ${OCP_AUTH_PROXY_MEMORY_LIMIT} | grep -oE $unit_regex || echo "")
+
+    if [ "$unit" = "G" ] || [ "$unit" = "g" ]; then
+        ((num = num * ${BYTES_PER_GIG})) # enables math to work out for odd Gi
+    elif [ "$unit" = "M" ]  || [ "$unit" = "m" ]; then
+        ((num = num * ${BYTES_PER_MEG})) # enables math to work out for odd Gi
+    #else assume bytes
+    fi
+
+    if [ $num -lt $DEFAULT_MIN ] ; then
+        echo "$num is less then the default $(($DEFAULT_MIN / $BYTES_PER_MEG))m.  Setting to default."
+        num=$DEFAULT_MIN
+    fi
+
+    export NODE_OPTIONS="--max-old-space-size=$((num / ${BYTES_PER_MEG}))"
+
+else
+    echo "Unable to process the OCP_AUTH_PROXY_MEMORY_LIMIT: '${OCP_AUTH_PROXY_MEMORY_LIMIT}'.  It must be in the format of: /${regex}/"
+    exit 1
```

Comment:
The run script sets a default instead of letting it run unbounded, which was the alternative.  I thought limiting was a better option.  It should only fail if a user provides a memory limit that has only alpha characters.